cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -340,3 +399,277 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration, volumes []v1.Vo
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.APIServerKubeletClientCertName),
+								},
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.APIServerKubeletClientKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.ServiceAccountKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.ServiceAccountPublicKeyName),
+								},
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.ServiceAccountPrivateKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.FrontProxyCACertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.FrontProxyCACertName),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func schedulerProjectedVolume() v1.Volume {
+	return v1.Volume{
+		Name: "k8s",
+		VolumeSource: v1.VolumeSource{
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: []v1.VolumeProjection{
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.SchedulerKubeConfigFileName,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func controllerManagerProjectedVolume() v1.Volume {
+	return v1.Volume{
+		Name: "k8s",
+		VolumeSource: v1.VolumeSource{
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: []v1.VolumeProjection{
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.ControllerManagerKubeConfigFileName,
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.CACertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.CACertName),
+								},
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.CAKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.ServiceAccountKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.ServiceAccountPrivateKeyName),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createTLSSecrets(client *clientset.Clientset) error {
+	if err := createTLSSecretFromFiles(
```

Comment:
please make an array of all the secrets that should be created, then loop through them and create them in the loop

Then we don't need the different funcs for controller-manager/scheduler either